### ISSUE TITLE
support non-intrusive pipeline parallelism

### ISSUE THREAD

**Issue by shifangx** [ByteDance-Seed/VeOmni#238]
### Feature request

In the paper [VeOmni](https://arxiv.org/abs/2508.02317), it is mentioned that VeOmni plan to support non-intrusive pipeline parallelism.
Is this feature currently under development? Could you share the design document? Approximately when will it be released?
> In future work, we plan to extend VeOmni to support non-intrusive pipeline parallelism, ...

### Motivation

Hope pipeline parallelism can be supported.

### Your contribution

I’d be happy to help if needed.

**Comment by Luosuu** [comment]
Hi @shifangx ,
we plan to support torch native pipeline parallelism but this is not our current priority. 
Mostly because we find fsdp2+ep can fit most of the cases and it is simpler to handle

**Comment by shifangx** [comment]
> Hi [@shifangx](https://github.com/shifangx) , we plan to support torch native pipeline parallelism but this is not our current priority. Mostly because we find fsdp2+ep can fit most of the cases and it is simpler to handle

Hi, @Luosuu, do you mean this pipeline? https://docs.pytorch.org/docs/stable/distributed.pipelining.html
Pipeline parallelism will be helpful for large models, such as models with 1T parameters.

**Comment by Luosuu** [comment]
> > Hi [@shifangx](https://github.com/shifangx) , we plan to support torch native pipeline parallelism but this is not our current priority. Mostly because we find fsdp2+ep can fit most of the cases and it is simpler to handle
> 
> Hi, @Luosuu, do you mean this pipeline? https://docs.pytorch.org/docs/stable/distributed.pipelining.html
> Pipeline parallelism will be helpful for large models, such as models with 1T parameters.

Yes. We have tried DeepSeek-671B/Qwen3-235B-A22B training with EP+FSDP2 and it is quite efficient. I do not quite see the urgency of pipeline parallelism yet

**Comment by piyifan123** [comment]
+1 to what @Luosuu said on fsdp2 + ep.

torch pp interface is still in alpha stage as of 2.9.1 https://docs.pytorch.org/docs/stable/distributed.pipelining.html do we know when it will be stable? since PP is very complex, developing over an alpha and constantly changing interface does not sound a good idea. we can re-evaluate once the torch support is stable in the future.

**Comment by shifangx** [comment]
> +1 to what [@Luosuu](https://github.com/Luosuu) said on fsdp2 + ep.
> 
> torch pp interface is still in alpha stage as of 2.9.1 https://docs.pytorch.org/docs/stable/distributed.pipelining.html do we know when it will be stable? since PP is very complex, developing over an alpha and constantly changing interface does not sound a good idea. we can re-evaluate once the torch support is stable in the future.

That was really helpful and gave me a lot of clarity — thanks again to walk me through it so thoroughly.

**Comment by shifangx** [comment]
> > > Hi [@shifangx](https://github.com/shifangx) , we plan to support torch native pipeline parallelism but this is not our current priority. Mostly because we find fsdp2+ep can fit most of the cases and it is simpler to handle
> > 
> > 
> > Hi, [@Luosuu](https://github.com/Luosuu), do you mean this pipeline? https://docs.pytorch.org/docs/stable/distributed.pipelining.html
> > Pipeline parallelism will be helpful for large models, such as models with 1T parameters.
> 
> Yes. We have tried DeepSeek-671B/Qwen3-235B-A22B training with EP+FSDP2 and it is quite efficient. I do not quite see the urgency of pipeline parallelism yet

As for DeepSeek-671B, could you share some performance number?
